klippa/src/gpos/mark_base_pos.rs

```diff
@@ -0,0 +1,96 @@
+//! impl subset() for MarkBasePos subtable
+use crate::{gpos::mark_array::collect_mark_record_varidx, CollectVariationIndices, Plan};
+use write_fonts::read::{collections::IntSet, tables::gpos::MarkBasePosFormat1, types::GlyphId};
+
+impl CollectVariationIndices for MarkBasePosFormat1<'_> {
+    fn collect_variation_indices(&self, plan: &Plan, varidx_set: &mut IntSet<u32>) {
+        let Ok(mark_coverage) = self.mark_coverage() else {
+            return;
+        };
+        let Ok(mark_array) = self.mark_array() else {
+            return;
+        };
+
+        let glyph_set = &plan.glyphset_gsub;
+        let mark_array_data = mark_array.offset_data();
+        let mark_records = mark_array.mark_records();
+
+        let mut retained_mark_classes = IntSet::empty();
+        for i in mark_coverage
+            .iter()
+            .enumerate()
+            .filter(|&(_i, g)| glyph_set.contains(GlyphId::from(g)))
+            .map(|(i, _)| i)
```

Comment:
these could probably all  be `filter_map`?

```suggestion
            .filter_map(|&(_i, g)| glyph_set.contains(GlyphId::from(g)).then_some(i))
```